### PR TITLE
Feature - billing

### PR DESCRIPTION
**Pull request by joefitter**
* Added asru routes for updating billing and loading billing history

### CODE REVIEW

**Comment by joefitter** [comment]
WIP until
https://github.com/UKHomeOffice/asl-service/pull/176
https://github.com/UKHomeOffice/asl-components/pull/99
https://github.com/UKHomeOffice/asl-pages/pull/653
https://github.com/UKHomeOffice/asl-public-api/pull/207
https://github.com/UKHomeOffice/asl-internal-api/pull/46
https://github.com/UKHomeOffice/asl-workflow/pull/193
https://github.com/UKHomeOffice/asl-resolver/pull/122

**Comment by joefitter** [comment]
Billing notes:

*External*

External admin users have a licence-fees page accessed from the dashboard where they can see fees for current year.  PILs include revoked and transfers that happened in that year, figures reflect this but we should add a lot of test coverage around this.

Fees and dates are defined here: https://github.com/UKHomeOffice/asl-pages/pull/653/files#diff-d249ae431a0871ff9768ad3dac381134R1-R28 - this should be checked to make sure the figures are correct, also it will need a new entry adding for 2020 financial year.

Previous years can be accessed (if defined in config file above), if year is not found it will default to 2019.

*Internal*

As above, but non-transferred licences have a billing property which can be changed by asru users by expanding the column and adding a comment.

History of billable changes are shown if present - these are taken from workflow using model-tasks with new filters (action and status) - this method works fine for current financial year, but we perhaps need to look at how to filter billable licences for previous years. Currently billable is a default true boolean on the model, we may need to make a join table similar to pil transfers?

*internal all licences*

All licence fees accessed through footer link, includes similar overview to above but for all establishments, also includes transferred licences twice (if still billable) as I remember this being a requirement (to and from establishment both billed in transfer year)

All personal licences tab - only includes active licences, transferred licences are not doubled up, they probably should show twice for the from and to establishment but I couldn't work out the join magic required.

*To do*

* Investigate historical billing status
* Don't default to 2019 in internal ui routing - should be most recent entry or current financial year
* Include transferred licences in internal all licences list

*To merge*

* asl-permissions g2g
* asl-schema g2g
* asl-components g2g
* asl-workflow g2g
* asl-resolver needs asl-schema
* asl-internal-api needs asl-schema and asl-service
* asl-public-api needs asl-schema and asl-service
* asl-pages needs asl-components and asl-service
* asl-internal-ui needs asl-pages, asl-service and asl-components
* asl needs asl-pages, asl-service and asl-components

**Comment by lennym** [comment]
Now WIP until https://github.com/UKHomeOffice/asl-components/pull/101 and https://github.com/UKHomeOffice/asl-pages/pull/665

File: package.json [hunk 1]
```diff
@@ -13,8 +13,7 @@
     "build": "npm run build:js && npm run build:css",
     "build:js": "node --max_old_space_size=4096 node_modules/.bin/webpack",
     "prebuild:css": "mkdir -p ./public/css",
-    "build:css": "npm-sass ./assets/sass/app.scss > ./public/css/app.css",
-    "postinstall": "npm run build"
```
Comment (wheelsandcogs) [inline package.json]: is this redundant / deprecated?

File: pages/licence-fees/index.js [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+          numPels,
+          fees,
+          personal,
+          establishment,
+          total,
+          startDate,
+          endDate,
+          year: req.year
+        };
+      })
```
Comment (wheelsandcogs) [inline pages/licence-fees/index.js]:
```
.then({ json: { meta, data }} => {
  res.locals.static.fees = {
    ...pick(meta, ['startDate', 'endDate']),
    ...pick(data, ['fees', 'personal', 'establishment', 'total']),
    numPils: data.numberOfPils,
    numPels: data.numberOfPels,
    year: req.year
  };
})
```

Comment (lennym) [reply pages/licence-fees/index.js]: WHO ARE YOU AND WHAT HAVE YOU DONE WITH PHIL?!

Comment (wheelsandcogs) [reply pages/licence-fees/index.js]: haha trying to fill in for @joefitter

Comment (lennym) [reply pages/licence-fees/index.js]: Too far. Too. Far.

**Pull request merged**